### ISSUE TITLE
"cannot find package" build failure with newer versions of Golang unless `GO111MODULE=auto` is passed

### ISSUE THREAD

**Issue by Nephyrin** [NVIDIA/libnvidia-container#162]
Arch linux ships a new enough version of go that the go module changes cause the build to break by default (See [AUR package])

```
go build -o libnvidia-container-go.so -ldflags "-s -w" -buildmode=c-shared .
main.go:22:2: cannot find package "nvcgo/internal/cgroup" in any of:
        /usr/lib/go/src/nvcgo/internal/cgroup (from $GOROOT)
        /home/$USER/go/src/nvcgo/internal/cgroup (from $GOPATH)
make[2]: *** [Makefile:40: libnvidia-container-go.so] Error 1
```

The workaround is running make with `GO111MODULE=auto` to allow go to select between module and non-module mode.  `make GO111MODULE=auto` accomplishes this.

The Makefile should probably do this by default until all modules are converted to the new format.

[AUR package]: https://aur.archlinux.org/pkgbase/libnvidia-container

**Comment by elezar** [comment]
For our containerised builds, we use [go 1.17.1](https://github.com/NVIDIA/libnvidia-container/blob/master/mk/docker.mk#L17). Which version of go is used for the Arch linux builds?

**Comment by ich777** [comment]
> @ich777 could it be that you have GO111MODULE set to something in your environment already?

No, not at all, this is always a fresh build environment (Debian 11)
I've now also checked this:
```
root@da937bad3889:~# echo $GO111MODULE

root@da937bad3889:~#
```
and also in `env` is nothing set in terms of go of course except for the `PATH`
